system/Test/FeatureTestTrait.php

```diff
@@ -265,15 +268,24 @@ public function options(string $path, ?array $params = null)
      */
     protected function setupRequest(string $method, ?string $path = null): IncomingRequest
     {
-        $path    = URI::removeDotSegments($path);
-        $config  = config(App::class);
-        $request = Services::request($config, false);
+        $config = config(App::class);
+        $uri    = new SiteURI($config);
 
         // $path may have a query in it
-        $parts                   = explode('?', $path);
-        $_SERVER['QUERY_STRING'] = $parts[1] ?? '';
+        $path  = URI::removeDotSegments($path);
+        $parts = explode('?', $path);
+        $path  = $parts[0];
+        $query = $parts[1] ?? '';
+
+        $_SERVER['QUERY_STRING'] = $query;
```

Comment:
Any reason to handle this via `Superglobals`?

Reply:
No. Refactored.